File: 007/cg_007_easy.rb
# @param {Integer} x
# @return {Integer}
def reverse(x)
    x_string = x.to_s
    x_string = x_string[0] == "-" ? x_string[0] + x_string[1..-1].reverse : x_string.reverse
    x_i = x_string.to_i
    x_i = (x_i <= 2147483647 && x_i >= -2147483648) ? x_i : 0
end

File: 007/cg_007_easy.java
public class Solution {
    public int reverse(int x) {
        boolean neg = x < 0 ? true : false;
        StringBuilder sb = new StringBuilder(Integer.toString(neg ? x * -1 : x));
        String s = trimLeadingZero(sb.reverse().toString());
        int res = 0;
        
        try {
            res = !s.isEmpty() ? Integer.parseInt(s) : 0;
        }
        catch (Exception ex) {
            return res;
        }
        
        return neg ? res * -1 : res;
    }
    
    private String trimLeadingZero(String s) {
        int p = 0;
        
        while (p < s.length() && s.charAt(p) == '0') {
            p++;
        }
        
        return p < s.length() ? s.substring(p) : "";
    }
}
